tests/rbd/rbd_migration_2diff_ec_pool_ec.py

```diff
@@ -0,0 +1,146 @@
+            "size": "10G",
+        },
+    }
+    kw["config"].update(config)
+    log.info(kw["config"])
+    initial_rbd_config(**kw)
+    image2 = f'{config["ec_pool_config"]["image"]}'
+    ecpool2 = f'{config["ec_pool_config"]["pool"]}'
+    src_spec2 = ecpool2 + "/" + image2
+
+    config = {
+        "ec-pool-only": True,
+        "ec-pool-k-m": "3,2",
+        "do_not_create_image": True,
+        "crush-failure-domain": "osd",
+        "ec_pool_config": {
+            "pool": "rbd_ECpool3",
+            "data_pool": "data_pool_ec3",
+        },
+    }
+    kw["config"].update(config)
+    log.info(kw["config"])
+    initial_rbd_config(**kw)
+    ecpool3 = f'{config["ec_pool_config"]["pool"]}'
+    dest_spec1 = ecpool3 + "/" + image1
+    dest_spec2 = ecpool3 + "/" + image2
+
+    try:
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        with parallel() as p:
+            p.spawn(run_fio, image_name=image1, pool_name=ecpool1, client_node=client)
+            p.spawn(run_fio, image_name=image2, pool_name=ecpool2, client_node=client)
+
+        # Perpare the migration process from 2 source pool to destination pool
+        with parallel() as p:
+            p.spawn(rbd.migration_prepare, src_spec1, dest_spec1)
+            p.spawn(rbd.migration_prepare, src_spec2, dest_spec2)
+
+        # verify migration prepare
+        with parallel() as p:
+            p.spawn(verify_migration, rbd, dest_spec1)
+            p.spawn(verify_migration, rbd, dest_spec2)
+
+        # run io's on the destination image
+        with parallel() as p:
+            p.spawn(run_fio, image_name=image1, pool_name=ecpool3, client_node=client)
+            p.spawn(run_fio, image_name=image2, pool_name=ecpool3, client_node=client)
+
+        # Execute migration process
+        with parallel() as p:
+            p.spawn(rbd.migration_action, action="execute", dest_spec=dest_spec1)
+            p.spawn(rbd.migration_action, action="execute", dest_spec=dest_spec2)
+
+        # commit migration process
+        with parallel() as p:
+            p.spawn(rbd.migration_action, action="commit", dest_spec=dest_spec1)
+            p.spawn(rbd.migration_action, action="commit", dest_spec=dest_spec2)
+
+        # verify commit migration
+        with parallel() as p:
+            p.spawn(verify_migration_commit, rbd, ecpool3, image1)
+            p.spawn(verify_migration_commit, rbd, ecpool3, image2)
+
+        images = [image1, image2]
+        # Display image info and verify
+        for image in images:
+            image_info = rbd.image_info(ecpool3, image)
+            log.info(f"Image info : {image_info}")
+            if image not in image_info["name"]:
+                log.error(
+                    f"Image {image} is not listed in 'rbd image info' for pool {ecpool3}"
+                )
+                return 1
+
+        return 0
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
```

Comment:
can you please add pool clean up in `finally` clause?

Reply:
working on it.

Reply:
Done.